docs/content/index.md

```diff
@@ -2,60 +2,153 @@
+
+### Historical Nodes
+
+Historical nodes commonly form the backbone of a Druid cluster. Historical nodes download immutable segments locally and serve queries over those segments.
+The nodes have a shared nothing architecture and know how to load segments, drop segments, and serve queries on segments.
+
+### Broker Nodes
+
+Broker nodes are what clients and applications query to get data from Druid. Broker nodes are responsible for scattering queries and gathering and merging results.
+Broker nodes know what segments live where.
+
+### Coordinator Nodes
+
+Coordinator nodes manage segments on historical nodes in a cluster. Coordinator nodes tell historical nodes to load new segments, drop old segments, and move segments to load balance.
+
+### Real-time Processing Nodes
+
+Real-time processing in Druid can currently be done using standalone realtime nodes or using the indexing service. The real-time logic is common between these two services.
+Real-time processing involves ingesting data, indexing the data (creating segments), and handing segments off to historical nodes. Data is queryable as soon as it is
+ ingested by the realtime processing logic. The hand-off process is also lossless; data remains queryable throughout the entire process.
+
+### External Dependencies
+
+Druid has a couple of external dependencies for cluster operations.
+
+* **Distributed Coordination** Druid relies on a distributed coordination service for intra-cluster communication. The only supported distributed coordination service right now is Zookeeper.
 
-Druid is partially inspired by search infrastructure and creates mostly immutable views of data and stores the data in data structures highly optimized for aggregations and filters. A Druid cluster is composed of various types of nodes, each designed to do a small set of things very well.
+* **Metadata Storage** Druid relies on a metadata storage to store metadata about segments and configuration. Services that create segments write new entries to the metadata store
+  and the coordinator nodes monitor the metadata store to know when new data needs to be loaded or old data needs to be dropped. The metadata store is not
+  involved in the query path. MySQL and PostgreSQL are popular metadata stores.
 
-Druid vs…
-----------
+* **Deep Storage** Deep storage acts as a permanent backup of segments. Services that create segments upload segments to deep storage and historical nodes download
+segments from deep storage. Deep storage is not involved in the query path. S3 and HDFS are popular deep storages.
 
-* [Druid-vs-Impala-or-Shark](Druid-vs-Impala-or-Shark.html)
-* [Druid-vs-Redshift](Druid-vs-Redshift.html)
-* [Druid-vs-Vertica](Druid-vs-Vertica.html)
-* [Druid-vs-Cassandra](Druid-vs-Cassandra.html)
-* [Druid-vs-Hadoop](Druid-vs-Hadoop.html)
-* [Druid-vs-Spark](Druid-vs-Spark.html)
-* [Druid-vs-Elasticsearch](Druid-vs-Elasticsearch.html)
+### High Availability Characteristics
```

Comment:
Consider taking this section out and making "high availability" one of the "best attributes" points.